### PR TITLE
SLSA provenance attestation

### PR DESCRIPTION
**Pull request by johnbillion**
See https://github.com/johnbillion/action-wordpress-plugin-attestation/pull/12

### CODE REVIEW

**Comment by szepeviktor** [comment]
You want two things in one workflow ...

**Comment by johnbillion** [comment]
Yeah it's not always easy to determine whether an action is a composite

**Comment by szepeviktor** [comment]
It is a reusable one. It cannot live as a step.

> reusable workflows should be referenced at the top-level `jobs.*.uses' key, not within steps

**Comment by szepeviktor** [comment]
I think this level3 thing must be in your build process.
https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-and-reusable-workflows-to-achieve-slsa-v1-build-level-3

**Comment by johnbillion** [comment]
I think I rabbit-holed on the actions provided by the SLSA framework due to the isolation requirement for level 3. The annoying thing about using a reusable workflow for a GitHub build provenance attestation is that you then need to verify it using the name of the repo that contains the reusable workflow that created the attestation.

It looks like I need to read into this more.

File: .github/workflows/deploy-tag.yml [hunk 1]
```diff
@@ -43,9 +43,23 @@ jobs:
...
+        uses: johnbillion/action-wordpress-plugin-attestation@slsa
         with:
           version: ${{ needs.deploy.outputs.version }}
           zip-path: ${{ needs.deploy.outputs.zip-path }}
+          dry-run: true
+  slsa:
+    name: Generate SLSA provenance attestation
+    needs: attest
+    permissions:
+      contents: read
```
Comment (szepeviktor) [inline .github/workflows/deploy-tag.yml]:
```suggestion
      contents: write
```

Comment (johnbillion) [reply .github/workflows/deploy-tag.yml]: Annoying that the workflow needs to be granted write permission even though the condition to run that job isn't satisfied.

**Pull request closed without merging**